src/server/db_slice.cc

```diff
@@ -1338,4 +1365,49 @@ void DbSlice::ResetUpdateEvents() {
   events_.update = 0;
 }
 
+void DbSlice::TrackKeys(const facade::Connection::WeakRef& conn, const ArgSlice& keys) {
+  if (conn.IsExpired()) {
+    DVLOG(2) << "Connection expired, exiting TrackKey function.";
+    return;
+  }
+
+  DVLOG(2) << "Start tracking keys for client ID: " << conn.GetClientId()
+           << " with thread ID: " << conn.Thread();
+  for (auto key : keys) {
+    DVLOG(2) << "Inserting client ID " << conn.GetClientId()
+             << " into the tracking client set of key " << key;
+    client_tracking_map_[key].insert(conn);
+  }
+}
+
+void DbSlice::SendInvalidationTrackingMessage(const std::string_view& key) {
+  if (client_tracking_map_.find(key) != client_tracking_map_.end()) {
+    // notify all the clients.
```

Comment:
Some nits: 
You can pass string_view by value
You can store the it = find(key) to avoid reading again with [key]

Reply:
fixed